Sources/TuistAutomation/Simulator/SimulatorController.swift

```diff
@@ -0,0 +1,107 @@
+            .mapToString()
+            .collectOutput()
+            .asSingle()
+            .flatMap { output in
+                guard output.standardError.isEmpty else {
+                    return .error(SimulatorControllerError.simctlError(output.standardError))
+                }
+                do {
+                    let data = output.standardOutput.data(using: .utf8)!
+                    let json = try JSONSerialization.jsonObject(with: data, options: [])
+                    guard let dictionary = json as? [String: Any],
+                        let devicesJSON = dictionary["devices"] as? [String: [[String: Any]]] else {
+                        return .just([])
+                    }
+
+                    let devices = try devicesJSON.flatMap { (runtimeIdentifier, devicesJSON) -> [SimulatorDevice] in
+                        try devicesJSON.map { (deviceJSON) -> SimulatorDevice in
+                            var deviceJSON = deviceJSON
+                            deviceJSON["runtimeIdentifier"] = runtimeIdentifier
+                            let deviceJSONData = try JSONSerialization.data(withJSONObject: deviceJSON, options: [])
+                            return try self.jsonDecoder.decode(SimulatorDevice.self, from: deviceJSONData)
+                        }
+                    }
+
+                    return .just(devices)
+                } catch {
+                    return .error(error)
+                }
+            }
+    }
+
+    func runtimes() -> Single<[SimulatorRuntime]> {
+        System.shared.observable(["/usr/bin/xcrun", "simctl", "list", "runtimes", "--json"])
+            .mapToString()
+            .collectOutput()
+            .asSingle()
+            .flatMap { output in
+                guard output.standardError.isEmpty else {
+                    return .error(SimulatorControllerError.simctlError(output.standardError))
+                }
+                do {
+                    let data = output.standardOutput.data(using: .utf8)!
+                    let json = try JSONSerialization.jsonObject(with: data, options: [])
+                    guard let dictionary = json as? [String: Any],
+                        let runtimesJSON = dictionary["runtimes"] as? [Any] else {
+                        return .just([])
+                    }
+
+                    let runtimesData = try JSONSerialization.data(withJSONObject: runtimesJSON, options: [])
+                    let runtimes = try self.jsonDecoder.decode([SimulatorRuntime].self, from: runtimesData)
+                    return .just(runtimes)
+                } catch {
+                    return .error(error)
+                }
+            }
+    }
+
+    func devicesAndRuntimes() -> Single<[SimulatorDeviceAndRuntime]> {
+        runtimes()
+            .flatMap { (runtimes) -> Single<([SimulatorDevice], [SimulatorRuntime])> in
+                print(runtimes)
+                return self.devices().map { ($0, runtimes) }
```

Comment:
```suggestion
                return self.devices().map { (device: $0, runtimes: runtimes) }
```

Giving these labels might make the `map` function below easier to understand